xc7/tests/ff_sr_ce/CMakeLists.txt

```diff
@@ -1,9 +1,52 @@
 add_file_target(FILE ff_ce_sr.v SCANNER_TYPE verilog)
+add_file_target(FILE ff_ce_sr_testbench.v SCANNER_TYPE verilog)
 
-add_fpga_target(
-  NAME ff_ce_sr
-  BOARD basys3
-  INPUT_IO_FILE ../common/basys3.pcf
-  SOURCES ff_ce_sr.v
-  EXPLICIT_ADD_FILE_TARGET
-  )
+function(ff_ce_sr_test num_ff)
+    get_target_property_required(PYTHON3 env PYTHON3)
+    get_target_property(PYTHON3_TARGET env PYTHON3_TARGET)
+
+    add_custom_command(
+        OUTPUT ff_ce_sr_${num_ff}.v
+        COMMAND ${PYTHON3} ${CMAKE_CURRENT_SOURCE_DIR}/generate.py
+            --template ${CMAKE_CURRENT_SOURCE_DIR}/ff_ce_sr.v
+            --params "NUM_FF=${num_ff}" > ${CMAKE_CURRENT_BINARY_DIR}/ff_ce_sr_${num_ff}.v
+        DEPENDS ${PYTHON3} ${PYTHON3_TARGET} ff_ce_sr.v generate.py
+        )
+    add_file_target(FILE ff_ce_sr_${num_ff}.v GENERATED)
+
+    add_custom_command(
+        OUTPUT ff_ce_sr_${num_ff}_tb.v
+        COMMAND ${PYTHON3} ${CMAKE_CURRENT_SOURCE_DIR}/generate.py
+            --template ${CMAKE_CURRENT_SOURCE_DIR}/ff_ce_sr_testbench.v
+            --params "NUM_FF=${num_ff}" >
+            ${CMAKE_CURRENT_BINARY_DIR}/ff_ce_sr_${num_ff}_tb.v
+        DEPENDS ${PYTHON3} ${PYTHON3_TARGET} ff_ce_sr_testbench.v generate.py
+        )
+    add_file_target(FILE ff_ce_sr_${num_ff}_tb.v GENERATED)
+    get_file_target(TB_SOURCE_TARGET ff_ce_sr_testbench.v)
+    get_file_target(TB_TARGET ff_ce_sr_${num_ff}_tb.v)
+    get_target_property(INCLUDE_FILES ${TB_SOURCE_TARGET} INCLUDE_FILES)
+    set_target_properties(${TB_TARGET} PROPERTIES INCLUDE_FILES
+        ${INCLUDE_FILES})
+
+    add_fpga_target(
+        NAME ff_ce_sr_${num_ff}
+        BOARD basys3
```

Comment:
Will we want to run this on other boards? If so should the board and pcf file be parameters to this function?

Reply:
I think this is little value in running this on other boards, all it will do is inflate the CI runtime for little value.  In addition because of the ROI, each top file will vary based on pinouts.  Once the ROI is gone we can revisit this.